### PR TITLE
enchancement/poll-page-mui

### PR DESCRIPTION
**Pull request by ria0308**

### CODE REVIEW

File: src/components/pages/poll-page/PollPage.styles.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  width: '100%',
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  flexGrow: 1,
+};
+
+export const pollPageContent: SxProps<Theme> = {
+  display: 'flex',
+  flexDirection: 'row',
```
Comment (Markusplay) [inline src/components/pages/poll-page/PollPage.styles.ts:15]: row is default for flex

File: src/components/pages/poll-page/components/answers-sheet/AnswerSheet.style.ts
Comment (Markusplay) [inline src/components/pages/poll-page/components/answers-sheet/AnswerSheet.style.ts:1]: bro wtf, use breakpoints

File: src/components/pages/poll-page/components/answers-sheet/AnswerSheet.style.ts [hunk 2]
```diff
@@ -1,15 +1,135 @@
...
+  width: 'fit-content',
+  borderRadius: '8px',
+
+  [theme.breakpoints.down('desktop')]: {
+    width: '100%',
+    marginBottom: '38px',
+  },
+};
+
+export const questionName: SxProps<Theme> = {
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/answers-sheet/AnswerSheet.style.ts:131]: use typography

File: src/components/pages/poll-page/components/answers-sheet/AnswersSaved/AnswerSaved.style.ts
Comment (Markusplay) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSaved/AnswerSaved.style.ts:1]: toOtherPolls and toMain seems like duplidates, fix it

File: src/components/pages/poll-page/components/questions-list/QuestionsList.tsx [hunk 3]
```diff
@@ -1,57 +1,48 @@
...
+        <Box key={id}>
+          <PollCard
+            key={id}
+            numberOfQuestions={category.count}
+            isComment={category.questions[0].type === 'TEXT'}
+            questionNumber={1 + id}
+            question={category.name}
+            numberOfAnswered={progress[id]}
+            isActive={currentCategory === id}
+            onClick={() => {
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/questions-list/QuestionsList.tsx:38]: extract this logic to handler

File: src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import theme from '@/styles/theme';
+
+export const questionNumber: SxProps<Theme> = {
+  marginBottom: '16px',
+  fontSize: '16px',
+  color: 'white',
+};
+
+export const questionTitle: SxProps<Theme> = {
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts:11]: use typography

File: src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  marginBottom: '14px',
+  fontSize: '20px',
+  fontWeight: '700',
+  color: 'white',
+};
+
+export const questionDescription: SxProps<Theme> = {
+  marginBottom: '20px',
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts:19]: use typography

File: src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  fontSize: '14px',
+  color: 'white',
+  fontWeight: '500',
+};
+export const questionCriteria: SxProps<Theme> = {
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/single-question/SingleQuestion.style.ts:24]: use typography

File: src/components/pages/poll-page/components/single-question/SingleQuestion.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+const SingleQuestion: FC<QuestionProps> = ({ id, question, count }) => {
+  const isMobile = useMediaQuery(theme.breakpoints.down('desktop'));
+  const numberRowsTextArea = isMobile ? 8 : 4;
+  const questionNumber =
+    question.type === QuestionType.TEXT
+      ? `Відкрите питання`
+      : `Питання ${id + 1} / ${count}`;
+
+  return (
+    // className={styles['question']
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/single-question/SingleQuestion.tsx:54]: remove comments

**Review by Markusplay — CHANGES_REQUESTED** [review]

**Comment by Yevheniy2020** [comment]
delete legacy cards

**Comment by Yevheniy2020** [comment]
add mui cards

File: src/components/pages/poll-page/components/questions-list/QuestionsList.tsx [hunk 8]
```diff
@@ -1,57 +1,70 @@
...
+      setFilterProgress(parsedProgress);
+    }
+
+    if (!checkIfAllZeros(progress)) {
+      setFilterProgress(progress);
+    }
+  }, [progress]);
+
+  function checkIfAllZeros(arr: number[]) {
+    for (let i = 0; i < arr.length; i++) {
```
Comment (Markusplay) [inline src/components/pages/poll-page/components/questions-list/QuestionsList.tsx:35]: use `every` method in this case instead `for`

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 9]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
 };
 
+const setCollectAnswers = (answers: Answer[], values: FormikValues) => {
+  const resultAnswersMap = new Map(
+    answers.map(answer => [answer.questionId, answer]),
+  );
+
+  for (const [valueId, value] of Object.entries(values)) {
+    const existingAnswer = resultAnswersMap.get(valueId);
+    if (existingAnswer !== undefined) {
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:47]:
```suggestion
    if (existingAnswer) {
```

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 10]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
+        sendingStatus != SendingStatus.SUCCESS && (
+          <>
+            <Box
+              sx={sxStyles.toQuestionsList}
+              onClick={() => {
+                setQuestionsListOpened(true);
+              }}
+            >
               <ChevronLeftIcon style={{ height: '20px' }} />
+              <Typography sx={sxStyles.questionName}>
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:204]:
```suggestion
              <ChevronLeftIcon height="20px"/>
```

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 11]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
-                  onClick={(event: FormEvent<HTMLFormElement>) => {
-                    // TODO: refactor this shit
-                    const name = (event.target as HTMLFormElement).name;
-                    const value = (event.target as HTMLFormElement).value;
-                    if (name && value) {
-                      values[name] = String(value);
-                      answer(values);
+            <Box sx={sxStyles.answersWrapper}>
+              <Formik
+                validationSchema={createValidationSchema}
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:210]: There is no argument provided, how does this validation schema work

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 12]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
+                validateOnMount
+                validateOnChange
+                initialValues={initialValues}
+                enableReinitialize
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:214]: initialValues never change, why do we need `enableReinitialize`

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 13]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
-                          await PollAPI.createTeacherGrade(
-                            { answers },
-                            disciplineTeacherId,
-                          );
-                          setIsSendingStatus(SendingStatus.SUCCESS);
-                        } catch (error) {
-                          displayError(error);
-                          setIsSendingStatus(SendingStatus.ERROR);
-                        }
+                    className={styles['form']}
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:225]: as far as I know we agreed upon not using classNames and switched to `sx`, consider adding `Box` as a wrapper

File: src/store/poll-page/usePollStore.ts [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  setQuestionsListOpened: (newValue: boolean) =>
+    set({ isQuestionsListOpened: newValue }),
+  setIsValid: (newValue: boolean) => set({ isValid: newValue }),
+  setCurrentQuestions: (newValue: Category) =>
+    set({ currentQuestions: newValue }),
+  setAnswers: (newAnswers: Answer[]) => set({ answers: newAnswers }),
+  setCurrentCategory: (newCategory: number) =>
+    set({ currentCategory: newCategory }),
+  setIsSendingStatus: (newStatus: SendingStatus) =>
+    set({ sendingStatus: newStatus }),
```
Comment (kujo205) [inline src/store/poll-page/usePollStore.ts:61]: I don't think you need this setters, updating values in zustand could be done using setState to update the store

And overall using zustand might be an overkill here, u could get away using vanilla react Context

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx [hunk 15]
```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
...
+                  >
+                    {currentQuestions?.questions.map((question, key) => (
+                      <SingleQuestion
+                        key={key}
```
Comment (kujo205) [inline src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:229]:
```suggestion
                        key={question.id}
```
never use an index as s key

File: src/components/pages/poll-page/components/questions-list/QuestionsList.tsx [hunk 16]
```diff
@@ -1,56 +1,91 @@
...
-  setCurrent,
-  setQuestionsListStatus,
-}) => {
+function checkIfAllZeros(arr: number[]) {
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] !== 0) {
+      return false;
+    }
+  }
+  return true;
```
Comment (kujo205) [inline src/components/pages/poll-page/components/questions-list/QuestionsList.tsx:22]:
```suggestion
  return arr.every(num=>num===0)
```

File: src/components/pages/poll-page/components/questions-list/QuestionsList.tsx [hunk 17]
```diff
@@ -1,56 +1,91 @@
...
+}
+
+function uniteArrs(arg1: number[], arg2: number[]): number[] {
```
Comment (kujo205) [inline src/components/pages/poll-page/components/questions-list/QuestionsList.tsx:25]: Well, naming is bad, I could'nt figure out purpose og this function

**Review by kujo205 — CHANGES_REQUESTED** [review]

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx (hunk 13 above)
Comment (Yevheniy2020) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:225]: doesn`t work

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx (hunk 15 above)
Comment (Yevheniy2020) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:229]: why? and look here 
image

Comment (kujo205) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:227]: image
[docs](https://legacy.reactjs.org/docs/lists-and-keys.html)

File: src/store/poll-page/usePollStore.ts (hunk 14 above)
Comment (kujo205) [reply src/store/poll-page/usePollStore.ts:61]: simply replace setters with `setState` in places where u use setters and remove setters, no need to remove `zustand`

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx (hunk 13 above)
Comment (kujo205) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:223]: try one more time)
or screw it, I don`t care

File: src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx (hunk 11 above)
Comment (kujo205) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:209]: image
No, it doesn't, an error comes from the backend

Comment (kujo205) [reply src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx:209]: Outsorce creteValidattioinSchena in s separate file pls

**Review by kujo205 — APPROVED** [review]

**Review by Yevheniy2020 — APPROVED** [review]

**Review by Markusplay — APPROVED** [review]

**Pull request merged**